addons/folo/common/src/main/java/org/commonjava/indy/folo/data/FoloRecordCassandra.java

```diff
@@ -0,0 +1,433 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.UriInfo;
+import java.net.MalformedURLException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+@FoloStoreToCassandra
+public class FoloRecordCassandra implements FoloRecord,StartupAction {
+
+    private final static String DOWNLOADS = "DOWNLOAD";
+    private final static String UPLOADS = "UPLOAD";
+
+    private final Logger logger = LoggerFactory.getLogger( this.getClass() );
+
+    @Inject
+    CassandraClient cassandraClient;
+
+    @Inject
+    FoloConfig config;
+
+    @Context
+    UriInfo uriInfo;
+
+    @FoloInprogressCache
+    @Inject
+    private CacheHandle<TrackedContentEntry, TrackedContentEntry> inProgressRecords;
+
+    @FoloSealedCache
+    @Inject
+    private CacheHandle<TrackingKey, TrackedContent> sealedRecords;
+
+    @Inject
+    private FoloBackupListener foloBackupListener;
+
+    @Inject
+    private FoloExpirationWarningListener expirationWarningListener;
+
+
+
+    private Session session;
+    private Mapper<DtxBuildRecord> buildsMapper;
+    private Mapper<DtxTrackingRecord> trackingMapper;
+
+    private PreparedStatement getTrackingRecordByBuildIdAndPath;
+    private PreparedStatement getTrackingRecordBySealed;
+    private PreparedStatement getTrackingRecordsByTrackingKey;
+
+
+    private static String createFoloBuildsTable( String keyspace )
+    {
+        return "CREATE TABLE IF NOT EXISTS " + keyspace + ".builds ("
+                + "id text,"
+                + "sealed boolean,"
+                + "started timestamp,"
+                + "finished timestamp,"
+                + "PRIMARY KEY (id)"
+                + ");";
+    }
+
+    private static String createFoloRecordsTable( String keyspace )
+    {
+        return "CREATE TABLE IF NOT EXISTS " + keyspace + ".records ("
+                + "tracking_key text,"
+                + "sealed boolean,"
+                + "store_key text,"
+                + "access_channel text,"
+                + "path text,"
+                + "origin_url text,"
+                + "local_url text,"
+                + "store_effect text,"
+                + "md5 text,"
+                + "sha256 text,"
+                + "sha1 text,"
+                + "size bigint,"
+                + "timestamps set<bigint>,"
+                + "PRIMARY KEY ((tracking_key),path)"
+                + ");";
+    }
+
+    private static String createFoloKeyspace( String keyspace)
+    {
+        return "CREATE KEYSPACE IF NOT EXISTS " + keyspace
+                + " WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':3 };";
+    }
+
+
+    private static String createFoloPathIdx(String  keyspace) {
+        return "CREATE INDEX IF NOT EXISTS path_idx ON " + keyspace + ".records (path);";
```

Comment:
I see 'path' is in primary key. so we do not need index here.

Reply:
you are. right , we. don't need. it. removed on next commit.